### DISCUSSION TITLE
Searching issue

### DISCUSSION THREAD

**Discussion by AhmadMujah** [pimcore/pimcore#12246] [Q&A / Support] [answered]
Hello

I'm new to pimcore, I installed a version on my Ubuntu 20 machine and I'm trying things out

My main usage for pimcore will be assets management to store them, classify them and search through them

I uploaded a bunch of assets ( images, videos, pdf, etc ... ) and I'm trying the search feature but it's not returning anything !

I tried searching by providing one the half name or full name or anything but nothing is working

![image](https://user-images.githubusercontent.com/101082885/170270564-25e0eca4-3758-42c2-b823-a8f729c6f4dd.png)

Am I doing something wrong ?

My main goal is to be able to search with keywords, meta and names

Thanks in advance !

**Comment by kingjia90** [comment, marked as answer]
Please try to run the `bin/console pimcore:search-backend-reindex` command

**Reply by AhmadMujah** [reply]
It worked, Should I do that each time I upload a file or modify something ?

**Reply by kingjia90** [reply]
i am glad to hear that it worked!
As far as i know, Pimcore should be automatically doing this indexing whenever an element is added/updated/deleted.

This command is just a "plan B" when thing somehow didn't go right and/or in edge cases eg. when there are some major changes like upgrade/migration or there was an error causing troubles to the indexing, db table corruption etc... so there is no need nor recommendation to make a cronjob or run it every time with an event like uploading.

**Reply by AhmadMujah** [reply]
Each time that I'm uploading something new, I have to re-execute the command 
Is there anyway to know why pimcore isn't doing this by itself ?

**Reply by kingjia90** [reply]
I will double-check if there's a problem in the latest releases, coincidentally there are 3 similar reports in the discussion area fixed by the same command, so it could be a bug 😟

**Reply by fashxp** [reply]
do you have setup the Pimcore maintenance properly? see also https://pimcore.com/docs/pimcore/current/Development_Documentation/Getting_Started/Installation.html#page_5-Maintenance-Cron-Job

**Reply by AhmadMujah** [reply]
I have set the cron jobs correctly on Ubuntu but still I need to execute the command in order to search correctly 
I have also re-installed the software to make sure that all is correct

**Reply by fashxp** [reply]
can you have a look at the log files and check, if pimcore is updating search index as soon as an element is saved?
